### ISSUE TITLE
ListenerCommand improper JSON serialisation

### ISSUE THREAD

**Issue by consultantleon** [Azure/azure-relay-java#60]
We found that the code to serialize the ListenerCommand toJsonString() method does not do a proper JSON serialization.
If any of the values serialized contains \ or " characters (or other special characters that JSON values should escape), the result is an invalid JSON serialization.
I'd suggest not to use JSONObject in the code (seems not to be a JSON de/serializer of of choice), but instead use proven JSON mapper solution from jackson/fasterxml.
At least never attempt to d.i.y. strings to JSON as it easily goes wrong as our case proved.

In below code we ran into troubles if HTTP headers contain a " character.

```
String toJsonString() {
			StringBuilder builder = new StringBuilder("{\"" + HybridConnectionConstants.Actions.RESPONSE + "\":{");
			List<String> fields = new ArrayList<String>();

			if (this.requestId != null) {
				fields.add("\"requestId\":\"" + this.requestId + "\"");
			}
			
			fields.add("\"body\":" + ((this.body) ? "true" : "false"));
			fields.add("\"statusCode\":" + this.statusCode);

			if (this.statusDescription != null) {
				fields.add("\"statusDescription\":\"" + this.statusDescription + "\"");
			}

			if (this.responseHeaders != null && !this.responseHeaders.isEmpty()) {
				StringBuilder headerBuilder = new StringBuilder("\"responseHeaders\":{");
				this.responseHeaders.forEach((key, val) -> {
					headerBuilder.append("\"" + key + "\":\"" + val + "\",");
				});
				headerBuilder.replace(headerBuilder.length() - 1, headerBuilder.length(), "}");
				fields.add(headerBuilder.toString());
			}
			
			builder.append(String.join(",", fields)).append("}}");
			return builder.toString();
		}
	}
```

We've done a quick internal patch to work around this bug using the JSONObject to do the serialization:
```
 if (this.responseHeaders != null && !this.responseHeaders.isEmpty()) {
                StringBuilder headerBuilder = new StringBuilder("\"responseHeaders\":");
                headerBuilder.append(new JSONObject(responseHeaders).toString());
                fields.add(headerBuilder.toString());
            }
```
(still we'd prefer to see this code use jackson / fasterxml)

**Comment by jfggdl** [comment]
@consultantleon Thank you for reporting this issue. We will work on this bug. The fix will probably not use external libraries to avoid additional dependencies.

**Comment by consultantleon** [comment]
I personally thank that you should use proven technologies to un/Marshall json. You can always repackage Jackson libraries within your own azure package structure? Doing string concats and replaces is error prone, handling of special eg utf characters and escape sequences should be done correctly and compliant with the json specs! Vulnerabilities should be tracked and dependencies updated accordingly.
Regards
Leon

**Comment by consultantleon** [comment]
Why is this blatant bug not fixed on 0.0.3? We constantly need to fork our own version to be able to use this library , currently only due to this bug not being fixed

**Comment by dlstucki** [comment]
It appears the same issue may be possible with statusDescription because the user also can set its value.
``` java
	if (this.statusDescription != null) {
		fields.add("\"statusDescription\":\"" + this.statusDescription + "\"");
	}
```

**Comment by bainian12345** [comment]
A separate PR was made to address the statusDescription. Please see PR #69 .

**Comment by bainian12345** [comment]
Fixed in version 0.0.3.
